pkg/ipam/ip.go

```diff
@@ -15,7 +16,7 @@ type IPRange struct {
 }
 
 func (a IP) Equal(b IP) bool {
-	return a == b
+	return net.ParseIP(string(a)).Equal(net.ParseIP(string(b)))
```

Comment:
```suggestion
	return a == b || net.ParseIP(string(a)).Equal(net.ParseIP(string(b)))
```

Reply:
Where is `Equal()` used? Is there any chance that `net.ParseIP(string(a))` may return nil?

Reply:
![image](https://user-images.githubusercontent.com/47097611/205865374-a17737a4-0029-4044-b4d8-442d7de6a366.png)

Reply:
in splitIPRangeList